src/libraries/CalldataDecoder.sol

```diff
@@ -17,6 +17,14 @@ library CalldataDecoder {
     /// error SliceOutOfBounds();
     uint256 constant SLICE_ERROR_SELECTOR = 0x3b99b53d;
 
+    /// @notice Removes the selector from the calldata and returns the encoded params.
+    function removeSelector(bytes calldata data) internal pure returns (bytes calldata params) {
```

Comment:
should you check that data.length > 4 here?
or can this be trusted?

Reply:
Yeah thats a good call, it shouldn't be because it would fail before this assuming the entrypoint code doesn't change, but probably a good addition?

Reply:
ah yeah so data.length is a full word btw so idt this is possible